bgapi/module.py

```diff
@@ -370,28 +438,37 @@ def get_module_info(self, timeout=0.5):
         return self._module_info
 
     def get_ble_address(self, timeout=1):
-        self.start_procedure(GET_ADDRESS)
-        self._api.ble_cmd_system_address_get()
-        if not self.wait_for_procedure(timeout=timeout):
-            raise BlueGigaModuleException("No response to get_ble_address!")
-        else:
-            return self.address
+        with self.procedure_call(GET_ADDRESS, timeout) as handle:
+            self._api.ble_cmd_system_address_get()
+        return self.address
 
     def reset_ble_state(self):
         """ Disconnect, End Procedure, and Disable Advertising """
         self._api.ble_cmd_gap_set_mode(gap_discoverable_mode['gap_non_discoverable'],
                                        gap_connectable_mode['gap_non_connectable'])
-        for i in range(8):
-            self.disconnect(i)
+
+        toDisconnect = set(self.connections.keys())
```

Comment:
When using this library for the BLE112 on USB, the BGAPI soft reset is not available.  When a python script dies while still connected to a remote device, the BLE112 will stay connected to the remote device.  When a new python script is started it will be unaware of the existing connections, which is why I use reset_ble_state() at the start of most of my scripts.  I could see what you've done here being a new method like "disconnect_all" where it intelligently disconnects from known connections.